src/main/java/appeng/items/contents/CellUpgrades.java

```diff
@@ -37,8 +37,8 @@ public CellUpgrades( final ItemStack is, final int upgrades )
 	}
 
 	@Override
-	public void markDirty()
-	{
-		this.writeToNBT( Platform.openNbtData( this.is ), "upgrades" );
-	}
+    protected void onContentsChanged(int slot)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') '(' is not followed by whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') ')' is not preceded with whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)